### ISSUE TITLE
Single GPU training still underperforms after learning rate and grad accumulation adjustment (follow-up to #7)

### ISSUE THREAD

**Issue by RyanPham19092002** [Tsinghua-MARS-Lab/TrackOcc#9]
Hi @zgchen33  ,

This is a follow-up to issue #7  (Significant performance drop when training from scratch on a single GPU), which contains my initial config, commands, and full screenshots for context. You can easily review all the details and metric images in that ticket.

Since your reply, I have:
- Reduced the learning rate proportional to the batch size change
- Used gradient accumulation to simulate 8-GPU training

Experiment 1: Gradient Accumulation
I manually implemented gradient accumulation to simulate the 8-GPU effective batch size, keeping the original learning rate at 2e-4.

Config I modified:
optimizer_config = dict(
cumulative_iters=8, grad_clip=dict(max_norm=35, norm_type=2))

Result summary: PQ4D = 0.169, AQ_overall = 0.109, iou_mean = 0.262

<img width="1180" height="159" alt="Image" src="https://github.com/user-attachments/assets/018f71a2-1c56-4e8f-84c2-b653c780d98d" />

Experiment 2: Learning Rate Adjustment
I ran the model on a single GPU (without gradient accumulation) and scaled down the learning rate by a factor of sqrt(8), changing it from 2e-4 to 5e-5.

Result summary: PQ4D = 0.161, AQ_overall = 0.106, iou_mean = 0.243

<img width="1189" height="161" alt="Image" src="https://github.com/user-attachments/assets/327c0b5e-e767-48fd-a860-b2ec23f02871" />

However, after retraining and validating, the performance gap with respect to the official weights remains large—the metrics do not reach the pretrained values, as shown again in the attached screenshots below. The training logs and detailed experiment setup remain unchanged, except for the two tweaks above.

Could you please advise if there are other configuration options, training parameters, or troubleshooting tips for better single-GPU convergence and results? Any advice to close the gap would be greatly appreciated.

(For all code, configs, and images: please see ticket #7 )

**Comment by zgchen33** [comment]
Hi, @RyanPham19092002 
Sorry, I don’t have further suggestions at this point. You might want to see if it’s possible to run on 8 GPUs to match the original setup.
